pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -140,6 +140,9 @@ const (
 	// StateAvailable means the host can be consumed
 	StateAvailable ProvisioningState = "available"
 
+	// StatePreparing means we are removing the existing configuration and set new configuration
+	StatePreparing ProvisioningState = "preparing"
```

Comment:
Maybe we discussed this early on in the review, so if so I apologize for raising the question again, but it's not entirely clear to me why we need a separate state. Isn't preparing the host just part of provisioning the image?

Reply:
Ironic ends up in the same state after manual cleaning as it was in before. So there's no way to tell if you've done it yet unless there is an explicit state transition in the Host. I agree this is a giant pain but it appears to be unavoidable without at least theoretically introducing race conditions.